### PR TITLE
required reviewers: remove reference to GA connectors

### LINKED ISSUE #31827: Require GL review for silver+ python connectors
## What

We currently have a required review from GL on GA connectors. Aside from moving away from release stages and into support/quality levels, we've received a request from @katmarkham to require GL's review for silver connectors, as GL has most context on these and we're receiving more contributions from other contributors external to GL.

As part of this we should make sure we only do this if it's a python source connector.

In addition, it's been noted that it's easy to skip over the required review check. It currently shows up as a "pending" status check if required reviewers haven't approved, and it is not marked as a required check.

We can swap the status to report as "failed" rather than "pending" so that it's more obvious, while keeping it to non-required to give engineers the autonomy to decide to skip it without having to /approve-and-merge.

### PR DESCRIPTION
**Pull request by alafanechere**
## What
Closes https://github.com/airbytehq/airbyte/issues/31827

* Minor change to the required reviewer logic: the SL threshold for a strategic connector is now 200
* Rename "important" connector to "strategic"
* Make gl-python required reviewers for python connectors only

### CODE REVIEW

**Comment by alafanechere** [comment]
Current dependencies on/for this PR:
* master
  * **PR #32151** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/32151" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/airbytehq/airbyte/32151?utm_source=stack-comment).

File: airbyte-ci/connectors/connector_ops/connector_ops/utils.py [hunk 1]
```diff
@@ -58,8 +58,8 @@ def download_catalog(catalog_url):
 METADATA_FILE_NAME = "metadata.yaml"
 ICON_FILE_NAME = "icon.svg"
 
-IMPORTANT_CONNECTOR_THRESHOLDS = {
-    "sl": 300,
+STRATEGIC_CONNECTOR_THRESHOLDS = {
+    "sl": 200,
```
Comment (bnchrch) [inline airbyte-ci/connectors/connector_ops/connector_ops/utils.py:62]: Im worried about this lowering 😬 

It seems like we may inadvertendly cause a lot of connectors to start failings tests

can you confirm this is correct?

**Review by bnchrch — APPROVED** [review]
Just one check to make, other than that this looks good.

Pre-approving

File: airbyte-ci/connectors/connector_ops/connector_ops/utils.py (hunk 1 above)
Comment (alafanechere) [reply airbyte-ci/connectors/connector_ops/connector_ops/utils.py:62]: I believe these thresholds are only used in the context of required reviewers so there should be no impact on the tests.

**Pull request merged**